server/app/auth/oidc/CiviformOidcLogoutActionBuilder.java

```diff
@@ -0,0 +1,95 @@
+package auth.oidc;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import auth.CiviFormProfileData;
+import com.google.common.collect.ImmutableMap;
+import com.nimbusds.openid.connect.sdk.LogoutRequest;
+import com.typesafe.config.Config;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Optional;
+import org.pac4j.core.context.WebContext;
+import org.pac4j.core.context.session.SessionStore;
+import org.pac4j.core.exception.TechnicalException;
+import org.pac4j.core.exception.http.RedirectionAction;
+import org.pac4j.core.profile.UserProfile;
+import org.pac4j.core.util.CommonHelper;
+import org.pac4j.core.util.HttpActionHelper;
+import org.pac4j.oidc.config.OidcConfiguration;
+import org.pac4j.oidc.logout.OidcLogoutActionBuilder;
+
+/**
+ * Custom OidcLogoutActionBuilder for CiviFormProfileData (since it extends CommonProfile, not
+ * OidcProfile). Also allows for divergence from the [oidc
+ * spec](https://openid.net/specs/openid-connect-rpinitiated-1_0.html) if your provider requires it
+ * (e.g. Auth0).
+ *
+ * <p>Does not provide the recommended id_token_hint, since these are not stored by the civiform
+ * profile.
+ *
+ * <p>Uses the post_logout_redirect_uri parameter by default, but allows overriding to a different
+ * value using the auth.oidc_post_logout_param config variable
+ *
+ * <p>If the oidc_logout_client_id_param config variable is set, also adds the client_id to the
+ * logout request.
+ */
+public final class CiviformOidcLogoutActionBuilder extends OidcLogoutActionBuilder {
+
+  private Optional<String> postLogoutRedirectParam;
+  private ImmutableMap<String, String> extraParams;
+
+  public CiviformOidcLogoutActionBuilder(
+      Config civiformConfiguration, OidcConfiguration oidcConfiguration, String clientID) {
+    super(oidcConfiguration);
+    checkNotNull(civiformConfiguration);
+    this.postLogoutRedirectParam =
+        getConfigurationValue(civiformConfiguration, "auth.oidc_post_logout_param");
+    Optional<String> clientIdParam =
+        getConfigurationValue(civiformConfiguration, "auth.oidc_logout_client_id_param");
+
+    if (clientIdParam.isPresent()) {
+      this.extraParams = ImmutableMap.of(clientIdParam.get(), clientID);
+    } else {
+      this.extraParams = ImmutableMap.of();
+    }
+  }
+
+  /** Helper function for retriving values from the application.conf, */
+  static final Optional<String> getConfigurationValue(Config civiformConfiguration, String name) {
```

Comment:
Not blocking: Can this be marked as `private`? Also, since the class is marked `final`, there's no need to add the specifier here.